src/puppetlabs/stockpile/queue.clj

```diff
@@ -108,12 +118,15 @@
       (fsync tmp false)
       (rename-durably tmp path sync-parent?)
       (catch Exception ex
-        ;; This approach will be revisited/revised after we discuss
-        ;; the alternatives a bit further.
         (try
-          (Files/deleteIfExists tmp)
-          (catch Exception ex
-            true))
+          (delete-if-exists tmp)
+          (catch Exception del-ex
+            (throw
+             (ex-info (str "unable to delete temp file " tmp " after error")
```

Comment:
We talked about this Monday and I know you're working up some changes here, I'll hold off on the rest of this review until that's done.